types/index.d.ts

```diff
@@ -57,18 +57,19 @@ export class Type {
   compareBuffers(buf1: Buffer, buf2: Buffer): number;
   constructor(schema: Schema, opts?: any);
   createResolver(type: any, opts?: any): any;  // TODO: opts not documented on wiki
-  decode(buf: Buffer, pos?: number, resolver?: any): any;
-  encode(val: any, buf: Buffer, pos?: number): void; 
-  fingerprint(algorithm?: any): any;
+  decode(buf: Buffer, pos?: number, resolver?: any): { value: any, offset: number};
```

Comment:
Decode: Good catch, I added that to my pullrequest as well, hoping all of this gets committed faster to the project.

Reply:
I'm OK either way. If @reuzel doesn't mind updating #175, I can merge this once `fingerprint`'s signature is changed; let me know what you prefer.

Reply:
done